### PR TITLE
Cleanup tidy for external deps

### LINKED ISSUE #10639: `tidy` package needs cleanup to work on external deps
In order to easily run `tidy` from dependencies' CI for https://github.com/servo/servo/issues/10636, we need a bit more Python packaging work: 

- [x] Declare an entry point, so that `tidy` can be run in the current directory by simply invoking `servo-tidy` on the command line. This'll involve adding something like the following to `setup()` in `setup.py` (note that the `tidy:scan` syntax might be wrong, though, so please test)

```
entry_points={
        'console_scripts': [
            'servo-tidy=tidy:scan',
        ],
    },
```

 - [x] Explicitly state the dependency on `flake8`, so it's installed with the package. See http://python-packaging.readthedocs.org/en/latest/dependencies.html for how. 

 - [x] Don't attempt to import wpt `tools.lint`  in https://github.com/servo/servo/blob/master/python/tidy/servo_tidy/tidy.py#L605 unless the `tests/wpt/web-platform-tests` directory exists in the current project. 

- [x] bump the minor version

@askeing, want to take this one?

### PR DESCRIPTION
**Pull request by askeing**
fix #10639

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10653)

### CODE REVIEW

File: python/tidy/servo_tidy/tidy.py [hunk 1]
```diff
@@ -601,11 +602,12 @@ def get_wpt_files(only_changed_files, progress):
 
 def check_wpt_lint_errors(files):
     wpt_working_dir = os.path.abspath(os.path.join(".", "tests", "wpt", "web-platform-tests"))
-    site.addsitedir(wpt_working_dir)
-    from tools.lint import lint
-    returncode = lint.lint(files)
-    if returncode:
-        yield ("WPT Lint Tool", "", "lint error(s) in Web Platform Tests: exit status {0}".format(returncode))
+    if os.path.isdir(wpt_working_dir):
```
Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Just being curious, why are we checking whether `wpt_working_dir` is a directory?

Comment (askeing) [inline python/tidy/servo_tidy/tidy.py]: The https://github.com/servo/servo/issues/10639#issue-148776716 has the requirement:
> Don't attempt to import wpt tools.lint in https://github.com/servo/servo/blob/master/python/tidy/servo_tidy/tidy.py#L605 unless the tests/wpt/web-platform-tests directory exists in the current project.

I think this requirement is for external projects (which do not have wpt), if there is no wpt folder, then skip the `tools.lint` import.

Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: Oh, right. We could've just checked `os.path.exists` :smile:

Comment (askeing) [inline python/tidy/servo_tidy/tidy.py]: I just want to make sure the path exists and is a directory :P
Do I have to modify it to `exists()`?

Comment (Wafflespeanut) [inline python/tidy/servo_tidy/tidy.py]: No, that's fine :)

**Comment by edunham** [comment]
Reviewed 1 of 3 files at r1, 1 of 3 files at r2, 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[python/tidy/HISTORY.rst, line 6 \[r4\]](https://reviewable.io:443/reviews/servo/servo/10653#-KFeH_yBiSqHBIq7j3aU:-KFeH_yCjPPIYCsqfYCW:-2110156913) ([raw file](https://github.com/servo/servo/blob/f5754b102c3abf2f7f4259f937353ebf972c9553/python/tidy/HISTORY.rst#L6)):*
This is OK for now, but it's generally clearer to specify exactly what changed than to just copy the title of the issue that a set of changes fixes. http://keepachangelog.com/ is better at explaining it than I am :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10653)*

**Comment by KiChjang** [comment]
@bors-servo retry
* #10457

**Pull request merged**